i3pystatus/wifionice.py

```diff
@@ -0,0 +1,219 @@
+
+    .. rubric:: Available formatters
+
+    * `{arrival_in}` - Time until arrival (in form "1h 12m" or "53m")
+    * `{arrival_time}` - Arrival time of train at the station (actual, if available, otherwise scheduled)
+    * `{delay}` - delay of train in minutes
+    * `{gps_lat}` - Current GPS latitude
+    * `{gps_lon}` - Current GPS longitude
+    * `{last_station_no}` - EVA number of the previous stop
+    * `{net_current}` - current state of network quality
+    * `{net_duration}` - how long until the next network quality change
+    * `{net_expected}` - next state of network quality
+    * `{next_platform}` - Platform number or name
+    * `{next_station_no}` - EVA number of the next stop
+    * `{next_station}` - Station name
+    * `{speed}` - Train speed in km/h
+    * `{train_no}` - Train number
+    * `{train_type}` - Train Type (probably always `ICE`)
+    """
+
+    final_destination = 'Endstation'
+    format_offtrain = None
+    format_ontrain = '{speed}km/h > {next_station} ({arrival_in}[ | {delay}])'
+    ice_status = {}
+    off_train_interval = 10
+    on_leftclick = 'open_url'
+    on_train_interval = 2
+    trip_info = {}
+    url_on_click = ''
+    wifi_adapters = ['wlan0']
+    wifi_names = ['WiFi@DB', 'WIFIonICE']
+
+    settings = (
+        ("final_destination", "Information text for 'final destination has been reached'"),
+        ("format_offtrain", "Formatter for 'not on a train' (module hidden if `None` - no formatters available)"),
+        ("format_ontrain", "Formatter for 'on a train'"),
+        ("off_train_interval", "time between updates if no train is detected"),
+        ("on_train_interval", "time between updates while on a train"),
+        ("url_on_click", "URL to open when left-clicking the module"),
+        ("wifi_adapters", "List of wifi adapters the module should consider "
+                          "when detecting if you are in a train. Set to `None` "
+                          "to disable that functionality."),
+        ("wifi_names", "List of Wifi network names that should be considered 'on a train'."),
+    )
+
+    def _format_time(self, seconds):
+        if seconds is None:
+            return "?"
+        seconds = int(seconds)
+        components = []
+        if seconds >= 3600:
+            hours = int(seconds / 3600)
+            seconds -= hours * 3600
+            components.append('{}h'.format(hours))
+        if seconds >= 60:
+            minutes = int(seconds / 60)
+            seconds -= minutes * 60
+            components.append('{}m'.format(minutes))
+        if not components:
+            components.append('now')
+        return " ".join(components)
+
+    def _check_wifi(self):
+        if self.wifi_adapters is None:
+            self.logger.debug('self.wifi_adapters is None')
+            return True
+
+        from basiciw import iwinfo
```

Comment:
With the default configuration, this import will cause a traceback unless basiciw is installed.

My personal preference is not to make the default configuration result in a traceback, and instead communicate a misconfiguration (or missing dependencies) via logging. However, a number of other i3pystatus modules result in a traceback if there are missing dependencies, and the default behavior of the `Status` class doesn't use a log file.

@enkore Do you have a specific preference for handling missing dependencies?

Reply:
I have modified this to print a warning to the logs if basiciw is not installed.